src/test/scala/chisel3/tests/OptionsPassingTest.scala

```diff
@@ -0,0 +1,66 @@
+package chisel3.tests
+
+import java.io.{ByteArrayOutputStream, File, PrintStream}
+
+import chisel3._
+import chisel3.tester._
+import org.scalatest._
+import treadle.{VerboseAnnotation, WriteVcdAnnotation}
+import chisel3.tester.experimental.TestOptionBuilder._
+
+class OptionsPassingTest extends FlatSpec with ChiselScalatestTester with Matchers {
+  behavior of "Testers2"
+
+  it should "write vcd output when passing in a WriteVcdAnnotation" in {
+    // TODO: this actually relies on total thread ordering
+
+    def shiftTest(in: UInt, out: UInt, clk: Clock, value: UInt) {
+      timescope {
+        in.poke(value)
+        clk.step(1)
+      }
+      clk.step(3)
+      out.expect(value)
+    }
+
+    test(new ShifterModule(UInt(8.W), 4))
+            .withAnnotations(Seq(WriteVcdAnnotation)) { c =>
+      fork { shiftTest(c.in, c.out, c.clock, 42.U) }
+      c.clock.step(1)
+      fork { shiftTest(c.in, c.out, c.clock, 43.U) }
+      c.clock.step(1)
+      fork { shiftTest(c.in, c.out, c.clock, 44.U) }
+      c.clock.step(1)
+      fork { shiftTest(c.in, c.out, c.clock, 45.U) }
+      c.clock.step(1)
+      fork { shiftTest(c.in, c.out, c.clock, 46.U) }
+      c.clock.step(1)
+      fork { shiftTest(c.in, c.out, c.clock, 47.U) }.join
+
+      val vcdFileName = "test_run_dir" +
+              File.separator +
+              sanitizeFileName(s"Testers2 should write vcd output when passing in a WriteVcdAnnotation") +
+              File.separator +
+              sanitizeFileName("ShifterModule") +
+              ".vcd"
+
+      val vcdFile = new File(vcdFileName)
+      println(s"vcd file name $vcdFileName ${vcdFile.getAbsolutePath}")
+      vcdFile.exists() should be (true)
+    }
+  }
+
+  it should "allow turning on verbose mode" in {
+    val outputStream = new ByteArrayOutputStream()
+    Console.withOut(new PrintStream(outputStream)) {
+      test(new ShifterModule(UInt(8.W), 4)).withAnnotations(Seq(VerboseAnnotation)) { c =>
+        c.clock.step(1)
+      }
+    }
+
+    val output = outputStream.toString
+
+    output.contains("Symbol table:") should be (true)
```

Comment:
Do we have anything we're particularly looking for in verbose mode or is this just going to be a [change detector](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html) test?
Especially `clock/prev`, is there any reason for that particular signal and formatting of the signal name? Or is this going to be brittle to however verbose mode is interpreted by Treadle?

Reply:
There is not a hard contract for what verbose mode is so it would be hard to make this an implementation test. Currently there is only treadle in the backend, when we introduce another backend, we will have to decide if there will be tests that depend on the backend selected.
I chose those two strings as unlikely to be found by accident. The `VerboseAnnotation` is treadle specific but it also something that I personally need to make sure I can pass into treadle when debugging. This test was made to confirm that process. I'd be happy to discuss how to make this all more rigorous

Reply:
Yeah, sounds like there isn't a great solution here (because there's no well-defined interface, but we still want this to be tested), so we can leave it as-is until this becomes a maintenance burden.